internal/command/status_test.go

```diff
@@ -13,93 +12,208 @@ import (
+			name:       "wildcard",
+			taskRunArg: []string{"*"},
+			expectedTaskIDs: []string{
+				"app1.build",
+				"app1.check",
+				"app1.test",
+				"app2.build",
+				"app2.check",
+				"app2.test",
+				"app3.build",
+				"app3.check",
+				"app3.test",
+				"app4.compile",
+				"app4.lint",
+			},
 		},
 		{
-			name:       "wildcard",
-			taskRunArg: "*",
+			name:       "appWildcard",
+			taskRunArg: []string{"app2.*"},
+			expectedTaskIDs: []string{
+				"app2.build",
+				"app2.check",
+				"app2.test",
+			},
+		},
+		{
+			name:       "AppnameWildcardAndTaskName",
+			taskRunArg: []string{"*.build"},
+			expectedTaskIDs: []string{
+				"app1.build",
+				"app2.build",
+				"app3.build",
+			},
 		},
+
 		{
-			name:       "taskSpec",
-			taskRunArg: fmt.Sprintf("%s.%s", app.Name, app.Tasks[0].Name),
+			name:       "appAndTaskWildcard",
+			taskRunArg: []string{"*.*"},
+			expectedTaskIDs: []string{
+				"app1.build",
+				"app1.check",
+				"app1.test",
+				"app2.build",
+				"app2.check",
+				"app2.test",
+				"app3.build",
+				"app3.check",
+				"app3.test",
+				"app4.compile",
+				"app4.lint",
+			},
 		},
+
 		{
-			name:       "taskSpecTaskWildcard",
-			taskRunArg: fmt.Sprintf("%s.%s", app.Name, "*"),
+			name:       "appName",
+			taskRunArg: []string{"app1"},
+			expectedTaskIDs: []string{
+				"app1.build",
+				"app1.check",
+				"app1.test",
+			},
 		},
 		{
-			name:       "taskSpecAppWildcard",
-			taskRunArg: fmt.Sprintf("%s.%s", "*", app.Tasks[0].Name),
+			name:       "specificTaskSpec",
+			taskRunArg: []string{"app4.lint"},
+			expectedTaskIDs: []string{
+				"app4.lint",
+			},
 		},
+
 		{
 			name:       "absPath",
-			taskRunArg: filepath.Dir(app.FilePath()),
+			taskRunArg: []string{filepath.Join(repoDir, "dir2", "app4")},
+			expectedTaskIDs: []string{
+				"app4.compile",
+				"app4.lint",
+			},
+		},
+		{
+			name:       "relPath",
+			taskRunArg: []string{filepath.Join("dir2", "app4")},
+			expectedTaskIDs: []string{
+				"app4.compile",
+				"app4.lint",
+			},
+		},
+		{
+			name:       "currentDirPath",
+			taskRunArg: []string{"."},
+			expectedTaskIDs: []string{
+				"app3.build",
+				"app3.check",
+				"app3.test",
+			},
+			preRun: func(t *testing.T) {
+				cwd, err := os.Getwd()
+				require.NoError(t, err)
+
+				require.NoError(t, os.Chdir("app3"))
+				t.Cleanup(func() {
+					require.NoError(t, os.Chdir(cwd))
+				})
+			},
+		},
+		{
+			name:       "parentDirPath",
+			taskRunArg: []string{".."},
+			expectedTaskIDs: []string{
+				"app3.build",
+				"app3.check",
+				"app3.test",
+			},
+			preRun: func(t *testing.T) {
+				childDir := filepath.Join(repoDir, "app3", uuid.New().String())
+				require.NoError(t, os.Mkdir(childDir, 0700))
+
+				t.Cleanup(func() {
+					require.NoError(t, os.RemoveAll(childDir))
+				})
+
+				cwd, err := os.Getwd()
+				require.NoError(t, err)
+
+				require.NoError(t, os.Chdir(childDir))
+				t.Cleanup(func() {
+					require.NoError(t, os.Chdir(cwd))
```

Comment:
Cleanup on a preparation function, very nice trick! I'm totally staling this idea.